Pixel Game/Assets/Scripts/Chest.cs

```diff
@@ -15,11 +15,19 @@ public class Chest : MonoBehaviour
     public Sprite openedEmptySprite;
     public Sprite openedWithLootSprite;
     private SpriteRenderer currentChestSprite;
+
+    private GameObject dataToPassGameObject;
     void Start()
     {
         mainCamera = GameObject.FindWithTag("MainCamera");
         chestInteractableScript = gameObject.GetComponent<Interactable>();
-        currentChestSprite = gameObject.GetComponent<SpriteRenderer>();      
+        currentChestSprite = gameObject.GetComponent<SpriteRenderer>();
+        dataToPassGameObject = GameObject.FindGameObjectWithTag("PassData");

```

Comment:
On line 25: I should use The DataToPass script as reference instead of gameobject